development/build/utils.js

```diff
@@ -1,32 +1,64 @@
+const semver = require('semver');
+const { version } = require('../../package.json');
+
+const BuildTypes = {
+  beta: 'beta',
+  flask: 'flask',
+  main: 'main',
+};
+
 /**
- * @returns {Object} An object with browser as key and next version of beta
- * as the value.  E.g. { firefox: '9.6.0.beta0', chrome: '9.6.0.1' }
+ * Map the current version to a format that is compatible with each browser.
+ *
+ * The given version number is assumed to be a SemVer version number. Additionally, if the version
+ * has a prerelease component, it is assumed to have the format "<build type>.<build version",
+ * where the build version is a positive integer.
+ *
+ * @param {string} currentVersion - The current version.
+ * @param {string[]} platforms - A list of browsers to generate versions for.
+ * @returns {Object} An object with the browser as the key and the browser-specific version object
+ * as the value.  For example, the version `9.6.0-beta.1` would return the object
+ * `{ firefox: { version: '9.6.0.beta1' }, chrome: { version: '9.6.0.1', version_name: 'beta' } }`.
  */
-function getNextBetaVersionMap(currentVersion, betaVersion, platforms) {
-  const [major, minor] = currentVersion.split('.');
+function getBrowserVersionMap(platforms) {
+  const major = semver.major(version);
+  const minor = semver.minor(version);
+  const patch = semver.patch(version);
+  const prerelease = semver.prerelease(version);
```

Comment:
Why not use `semver.parse(version)` here?

Reply:
I wasn't sure what that would do for us :thinking: I see that it returns a SemVer object, but that class didn't seem to have any documented `major`, `minor`, `patch`, or `prerelease` methods or properties.

Since you mentioned it, I did dig in and I found that it has these properties. But I'm not sure whether they're public API or not.

Reply:
```
const semverParse = require('semver/functions/parse')
```

That's listed as a usage example in [their readme](https://github.com/npm/node-semver#usage), so I can't imagine it isn't public. However, I'm fine with it either way!